tests/__init__.py

```diff
@@ -349,6 +349,43 @@ def test_thingy_delete_one(TestThingy, collection):
     assert TestThingy.find_one({"bar": "baz"}) is None
 
 
+def test_thingy_update_one(TestThingy, collection):
+    collection.insert_many(
+        [{"foo": "bar"}, {"bar": "qux"}, {"bar": "baz"}, {"bar": "baz"}]
+    )
+    for i in range(2):
+        updated = TestThingy.update_one({"bar": "baz"}, {"$set": {"bar": "baaz"}})
+        assert updated.acknowledged
+        assert updated.matched_count == 1
+        assert updated.modified_count == 1
+        if i == 0:
+            assert TestThingy.find_one({"bar": "baz"}) is not None
+        else:
+            assert TestThingy.find_one({"bar": "baz"}) is None
```

Comment:
I think this could be done better but I can't find something readable and light

Reply:
what are you testing exactly?

Reply:
The element `{"bar": "baz"}` exists multiple times.
The first update_one should affect only one of them, so if we `find_one({"bar": "baz"})`, one of them is still present on the first iteration.
On the second iteration, none of them exist any more

Reply:
Okay, that's wasn't obvious. I'd say we should do it sequentially following this though process so that it's easier to understand. :)